src/containers/Connect.svelte

```diff
@@ -43,16 +44,48 @@
     }
 
     // Start listening to ENABLED and DISABLED events and binds to the state
-    await marina.on('ENABLED', () => {
-      notifyEnabled();
-    });
+    listeners.push(
+      marina.on('ENABLED', () => {
+        notifyEnabled();
+      })
+    );
 
-    await marina.on('DISABLED', () => {
-      notifyDisabled();
-    });
+    listeners.push(
+      marina.on('DISABLED', () => {
+        notifyDisabled();
+      })
+    );
+
+    listeners.push(
+      marina.on('NEW_UTXO', (u: Utxo) => {
+        marinaStore.update((s: MarinaStore) => ({
+          ...s,
+          utxos: [...s.utxos, u],
```

Comment:
not clear how we can get the coins already in marina (ie before we start listening) withotu calling the getCoins?

Reply:
the `NEW_UTXO` event is **relative to the injected page** it means that "NEW" doesn't mean "new in marina" but "never sent to web page". So if we subscribe to `NEW_UTXO` event type, we'll receive first all the "existing" unspents directly. Maybe this is not clear in doc I think

Reply:
ah totally missed this feature. Yes will need to make it more clear